pkg/controller/atlasdatabaseuser/databaseuser.go

```diff
@@ -218,52 +208,59 @@ func filterScopeDeployments(user akov2.AtlasDatabaseUser, allDeploymentsInProjec
 	return deploymentsToCheck
 }
 
-func shouldUpdate(log *zap.SugaredLogger, atlasSpec *mongodbatlas.DatabaseUser, operatorDBUser akov2.AtlasDatabaseUser, currentPasswordResourceVersion string) (bool, error) {
-	matches, err := userMatchesSpec(log, atlasSpec, operatorDBUser.Spec)
+func shouldUpdate(log *zap.SugaredLogger, atlasUser *dbuser.User, operatorUser *akov2.AtlasDatabaseUser, currentPasswordResourceVersion string) (bool, error) {
+	matches, err := userMatchesSpec(log, &atlasUser.AtlasDatabaseUserSpec, &operatorUser.Spec)
 	if err != nil {
 		return false, err
 	}
 	if !matches {
 		return true, nil
 	}
 	// We need to check if the password has changed since the last time
-	passwordsChanged := operatorDBUser.Status.PasswordVersion != currentPasswordResourceVersion
+	passwordsChanged := operatorUser.Status.PasswordVersion != currentPasswordResourceVersion
 	if passwordsChanged {
 		log.Debug("Database User password has changed - making the request to Atlas")
 	}
 	return passwordsChanged, nil
 }
 
-// TODO move to a separate utils (reuse from deployments)
-func userMatchesSpec(log *zap.SugaredLogger, atlasSpec *mongodbatlas.DatabaseUser, operatorSpec akov2.AtlasDatabaseUserSpec) (bool, error) {
-	userMerged := mongodbatlas.DatabaseUser{}
-	if err := compat.JSONCopy(&userMerged, atlasSpec); err != nil {
+func userMatchesSpec(log *zap.SugaredLogger, atlasUsername, operatorUser *akov2.AtlasDatabaseUserSpec) (bool, error) {
+	operatorCopy, err := dbuser.Normalize(operatorUser.DeepCopy())
```

Comment:
DeepCopy'ing here this is actually fine 👍 We're deep copying as an internal implementation detail for normalization. Great job! 🎉